### PR TITLE
resolve: Fix resolution of empty paths passed from rustdoc

### PR DESCRIPTION
**Pull request by petrochenkov**
Fixes https://github.com/rust-lang/rust/pull/95337#issuecomment-1088426179

### CODE REVIEW

File: src/test/rustdoc-ui/intra-doc/global-path.stderr [hunk 1]
```diff
@@ -0,0 +1,10 @@
+warning: unresolved link to `::Unresolved`
+  --> $DIR/global-path.rs:6:6
+   |
+LL | /// [::Unresolved]
+   |      ^^^^^^^^^^^^ no item named `` in scope
```
Comment (rylev) [inline src/test/rustdoc-ui/intra-doc/global-path.stderr:5]: Unrelated to the changes here, this error message seems wrong. Should we open an issue to change the [diagnostic](https://github.com/rust-lang/rust/blob/7f3e2c0d374ba7982ef09499c7de6bb53866abf4/src/librustdoc/passes/collect_intra_doc_links.rs#L916) to check for empty paths?

**Comment by lopopolo** [comment]
Hi folks, I think I may be getting an ICE related to this PR: https://github.com/rust-lang/rust/issues/95712

File: src/test/rustdoc-ui/intra-doc/global-path.stderr (hunk 1 above)
Comment (petrochenkov) [reply src/test/rustdoc-ui/intra-doc/global-path.stderr:5]: Maybe.
I don't plan to work on this personally.

Comment (camelid) [reply src/test/rustdoc-ui/intra-doc/global-path.stderr:5]: I'm looking into it.

**Pull request merged**